frontend/views/containers/payments/PaymentsMixin.js

```diff
@@ -12,23 +12,70 @@ const PaymentsMixin: Object = {
     ...mapState(['currentGroupId']),
     ...mapGetters([
       'currentGroupState',
+      'dueDateForPeriod',
+      'groupPeriodPayments',
+      'groupSettings',
+      'groupSortedPeriodKeys',
       'ourUsername',
       'ourPayments',
-      'groupPeriodPayments',
-      'paymentHashesForPeriod',
-      'dueDateForPeriod'
+      'periodAfterPeriod',
+      'periodBeforePeriod',
+      'periodStampGivenDate',
+      'paymentHashesForPeriod'
     ])
   },
   methods: {
+    async historicalPeriodStampGivenPayment (payment) {
+      return await this.historicalPeriodStampGivenDate(payment.date)
+    },
+
+    // Oldest key first.
+    async getAllSortedPeriodKeys () {
+      const historicalPaymentPeriods = await this.getHistoricalPaymentPeriods()
+      return [
+        ...Object.keys(historicalPaymentPeriods).sort(),
+        ...this.groupSortedPeriodKeys
+      ]
+    },
+
+    async historicalPeriodStampGivenDate (givenDate: string | Date) {
+      return periodStampsForDate(givenDate, {
+        knownSortedStamps: await this.getAllSortedPeriodKeys(),
+        periodLength: this.groupSettings.distributionPeriodLength
+      }).current
+    },
+
+    async historicalPeriodBeforePeriod (periodStamp: string) {
+      return periodStampsForDate(periodStamp, {
+        knownSortedStamps: await this.getAllSortedPeriodKeys(),
+        periodLength: this.groupSettings.distributionPeriodLength
+      }).previous
+    },
+
+    async historicalPeriodAfterPeriod (periodStamp: string) {
+      return periodStampsForDate(periodStamp, {
+        knownSortedStamps: await this.getAllSortedPeriodKeys(),
+        periodLength: this.groupSettings.distributionPeriodLength
+      }).next
+    },
```

Comment:
Beautiful 😍

Reply:
I'm quite proud of this code, thank you! 😄